### PR TITLE
[Aptos Framework] Add more e2e tests for Staking

### PR DESCRIPTION
**Pull request by movekevin**
### Test Plan

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2592)

### CODE REVIEW

File: aptos-move/e2e-move-tests/src/lib.rs [hunk 1]
```diff
@@ -187,6 +197,121 @@ impl MoveHarness {
         self.run(txn)
     }
 
+    pub fn setup_staking(&mut self, account: &Account, initial_stake_amount: u64) {
```
Comment (wrwg) [inline aptos-move/e2e-move-tests/src/lib.rs:200]: I thought this landed already as I approved this but now I seem to get a 2nd chance to ask for something here: I suggest we move this into a different struct and module. One problem with the old infra is that it _a lot_ of stuff and was hard to navigate. Lets try to keep this small and structured. New usage could be like this:

```
let mut stake_harness = StakeHarness::new(harness); // move harness into stake_harness.
stake_harness.setup_staking(...);
stake_harness.harness.some_base_func(...); // can still use functions in the underlying harness
...
let mut harness = stake_harness.into_inner() 
```

Comment (wrwg) [reply aptos-move/e2e-move-tests/src/lib.rs:200]: > 

Another advantage is that you can start adding state specific to the functions in `StakeHarness`. Whether StakeHarness is the right name I don't know. Perhaps `AptosHarness` is better.

File: aptos-move/e2e-move-tests/tests/stake.rs [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+    // Validator should now be locked up.
+    let stake_pool = harness.get_stake_pool(&owner_address);
+    assert_eq!(
+        stake_pool.locked_until_secs,
+        harness.executor.get_block_time_seconds() + 7200
+    );
+
+    // Unlock stake.
+    assert_success!(harness.unlock_stake(&owner, stake_amount / 2));
```
Comment (igor-aptos) [inline aptos-move/e2e-move-tests/tests/stake.rs:42]: let's have unlocked stake and stake left be different, so you can confirm right amount is removed. i.e. 1/4 and 3/4 for example?

File: aptos-move/e2e-move-tests/tests/stake.rs [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+fn test_staking_rewards() {
+    let mut harness = MoveHarness::new();
+    enable_golden!(harness);
+    let validator_1 = harness.new_account_at(AccountAddress::from_hex_literal("0x123").unwrap());
+    let validator_2 = harness.new_account_at(AccountAddress::from_hex_literal("0x234").unwrap());
+    let validator_1_address = *validator_1.address();
+    let validator_2_address = *validator_2.address();
+
+    // Initialize the validators.
+    let rewards_per_epoch = 1141;
```
Comment (igor-aptos) [inline aptos-move/e2e-move-tests/tests/stake.rs:86]: can you put a formula (or comment the formula) for this magical number. (i.e. 0.1/epochs_in_a_year, etc)

**Review by igor-aptos — APPROVED** [review]
looks good to me, assuming you address/reply to @wrwg's comments

File: aptos-move/e2e-move-tests/tests/stake.rs (hunk 3 above)
Comment (movekevin) [reply aptos-move/e2e-move-tests/tests/stake.rs:86]: It's hard to get the formula as the configs (rewards rate, epoch duration, etc.) are buried deep in the genesis configurations. It'll take some code to dig it out. We can probably do this in a separate PR later.

File: aptos-move/e2e-move-tests/src/lib.rs (hunk 1 above)
Comment (movekevin) [reply aptos-move/e2e-move-tests/src/lib.rs]: Updated. I separated out the stake calls into a separate library in src/stake.rs. They don't need to store any state between calls for now so doesn't need to be an object like MoveHarness.

File: aptos-move/e2e-move-tests/tests/stake.rs (hunk 2 above)
Comment (movekevin) [reply aptos-move/e2e-move-tests/tests/stake.rs]: Updated.

**Pull request merged**